syntax = "proto3";

package vacancies;

option go_package = "./vacancies";

// protoc -I=vacancies --go-grpc_out=vacancies --go_out=vacancies vacancies/vacancies.proto

service Vacancies {
    rpc List (ListRequest) returns (ListResponse);
    rpc Get (GetRequest) returns (Vacancy);
    rpc Create (CreateRequest) returns (Vacancy);
    rpc Edit (Vacancy) returns (Vacancy);
}

message ListRequest{}

message Vacancy {
    int32 id = 1;
    string name = 2;
    string description  = 3;
    bool is_active = 4;
    string token  = 5;
}

message ListResponse {
    repeated Vacancy vacancies = 1;
}

message GetRequest {
    int32 id = 1;
}

message CreateRequest {
    string name = 1;
    string description  = 2;
    bool is_active = 3;
    string token = 4;
}
